File: FirstClass/problem33.m
#Ejemplo problem33 (12, 1, 1)
function problem33(frec,mag,fase)
max_plots=4; 
num_points=200; 
    if (or((length(mag)~=length(frec)), (length(fase)~=length(frec))))
    error('Error - frec, mag, y fase deben ser del mismo tamaño')
    endif

frec_min=min(frec);
step=2/(frec_min*num_points);
t=0:step:2*(1/frec_min);
i=1;
mi_limite=min(length(frec),max_plots);

while (i<=mi_limite)
    x=mag(i)*cos(2*pi*frec(i)*t+fase(i));
    mi_titulo= sprintf('Sinusoide con f=%d Hz.,mag=%4.2f, fase=%4.2f radianes',frec(i), mag(i), fase(i));
    subplot(mi_limite,1,i)
    plot(t,x);
    grid;
    title(mi_titulo);
    ylabel('Amplitud');
    i=i+1;
    xlabel('Tiempo (seg)');
    disp (i)
endwhile

endfunction

File: FirstClass/Python/problem_33.py
import numpy as np
import matplotlib.pyplot as plt
import math
if __name__ == "__main__":
    max_plots = 4
    num_points = 200
    magnitud = list(map(int, input('Agregue magnitud: \n').rstrip().split()))
    frecuencia = list(
        map(int, input('Agregue frecuencia: \n').rstrip().split()))
    fase = list(map(int, input('Agregue fase: \n').rstrip().split()))

    if (len(magnitud) != len(frecuencia) or len(frecuencia) != len(fase)):
        print('Dimensionalmente incorrecto')
        exit()

    # magnitud = np.array(magnitud)
    # frecuencia = np.array(frecuencia)
    # fase = np.array(fase)

    frec_min = np.min(frecuencia)
    step = 2 / (frec_min * num_points)
    t = np.linspace(0, 100, 200)
    i = 0
    mi_limite = min(len(frecuencia), max_plots)
    print(mi_limite)

    if mi_limite != 1:
        fig, axs = plt.subplots(mi_limite)
        while i < mi_limite:
            print(magnitud[i])
            x = magnitud[i] * np.cos(2 * np.pi * frecuencia[i] * t + fase[i])
            axs[i].plot(t, x)
            plt.grid()
            i += 1
        plt.show()
    else:
        fig, axs = plt.subplots()
        x = magnitud[0] * np.cos(2 * np.pi * frecuencia[0] * t + fase[0])
        axs.plot(t, x)
        plt.show()
